pkg/engine/task/task.go

```diff
@@ -0,0 +1,69 @@
+package task
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1alpha1"
+)
+
+// Tasker is an interface that represents any runnable task for an operator. This method is treated
+// as idempotent and will be called multiple times during the life-cycle of the plan execution.
+// Method returns a boolean, signalizing that the task has finished successfully, and an error.
+// An error can wrap the ErrFatalExecution for errors that can not be retried e.g. failed template
+// rendering. This will result in a v1alpha1.ExecutionFatalError in the plan execution status. A normal
+// error e.g. failure during accessing the API server will be treated  as "transient", meaning plan
+// execution engine  can retry it next time. Only a (true, nil) return value will be treated as successful
+// task execution.
+type Tasker interface {
+	Run(ctx Context) (bool, error)
+}
+
+// Available engine tasks kinds
+const (
+	ApplyTaskKind  = "Apply"
+	DeleteTaskKind = "Delete"
+	DummyTaskKind  = "Dummy"
+)
+
+var (
+	// ErrFatalExecution is a wrapper for the fatal engine task execution error
+	ErrFatalExecution = errors.New("")
```

Comment:
could this maybe say at least "Fatal error:"?

Reply:
this was advertised as a marker.  I'm not sure what adding text here will accomplish... if text is added we will need to change the error wrapping where "%wblah" is used for sure.

Reply:
it could still be marker with non-empty text, or not? 🤔 I assumed then it would end up being "Fatal error: blah"

Reply:
>  I was just proposing to do ErrFatalExecution = errors.New("Fatal error:") to get that text prepended to the messages where it is fatal

I had this exact text there for some time and removed it because we wrap the errors a few times and the end result just doesn't read good. So, currently what a user will see is:
```
PlanExecution: fatal error during task validation execution for operator version first-operator-zljnmj: unknown task kind Foo```

I felt that it reads better with _fatal_ at the beginning, but I can rearrange.

Reply:
Oh I see, yeah good point with the wrapping :)